### PR TITLE
feat(spanner): Add samples for Cloud Spanner Default Backup Schedules

### PR DESCRIPTION
**Pull request by Sally-Ye**
## Description

This PR adds code samples for cloud spanner default backup schedules using the following apis:
1. CreateInstance
2. UpdateInstance

## Checklist
- [X] I have followed [Contributing Guidelines from CONTRIBUTING.MD](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md)
- [X] **Tests** pass:   `go test -v ./..` (see [Testing](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#testing))
- [ ] **Code formatted**:   `gofmt` (see [Formatting](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#formatting))
- [X] **Vetting** pass:   `go vet` (see [Formatting](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#formatting))
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample
- [X] Please **merge** this PR for me once it is approved

### CODE REVIEW

File: spanner/spanner_snippets/spanner/spanner_create_instance_without_default_backup_schedule.go [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+			NodeCount:                 1,
+			Labels:                    map[string]string{"cloud_spanner_samples": "true"},
+			DefaultBackupScheduleType: instancepb.Instance_NONE,
+		},
+	})
+	if err != nil {
+		return fmt.Errorf("could not create instance %s: %w", fmt.Sprintf("projects/%s/instances/%s", projectID, instanceID), err)
+	}
+	// Wait for the instance creation to finish.
+	i, err := op.Wait(ctx)
```
Comment (telpirion) [inline spanner/spanner_snippets/spanner/spanner_create_instance_without_default_backup_schedule.go:52]: issue: use a more descriptive name for this variable. What type is `i`? if I wanted to look up reference documentation about it, what would I lookup?

File: spanner/spanner_snippets/spanner/spanner_update_instance_default_backup_schedule_type.go [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+	// projectID := "my-project-id"
+	// instanceID := "my-instance"
+	ctx := context.Background()
+	instanceAdmin, err := instance.NewInstanceAdminClient(ctx)
+	if err != nil {
+		return err
+	}
+	defer instanceAdmin.Close()
+
+	req := &instancepb.UpdateInstanceRequest{
```
Comment (telpirion) [inline spanner/spanner_snippets/spanner/spanner_update_instance_default_backup_schedule_type.go:38]: issue: update operations are a commonly confused task for developers. Provide a comment above the request object that describes:

1. What field in the resource is being updated.
2. Why the FieldMask must be included.

File: spanner/spanner_snippets/spanner/spanner_update_instance_default_backup_schedule_type.go [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+	"context"
+	"fmt"
+	"io"
+
+	instance "cloud.google.com/go/spanner/admin/instance/apiv1"
+	"cloud.google.com/go/spanner/admin/instance/apiv1/instancepb"
+	"google.golang.org/genproto/protobuf/field_mask"
+)
+
+func updateInstanceDefaultBackupScheduleType(w io.Writer, projectID, instanceID string) error {
```
Comment (telpirion) [inline spanner/spanner_snippets/spanner/spanner_update_instance_default_backup_schedule_type.go:28]: issue: provide a short Godoc-style comment that explains what this function does.

File: spanner/spanner_snippets/spanner/spanner_create_instance_without_default_backup_schedule.go [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+import (
+	"context"
+	"fmt"
+	"io"
+
+	instance "cloud.google.com/go/spanner/admin/instance/apiv1"
+	"cloud.google.com/go/spanner/admin/instance/apiv1/instancepb"
+)
+
+func createInstanceWithoutDefaultBackupSchedule(w io.Writer, projectID, instanceID string) error {
```
Comment (telpirion) [inline spanner/spanner_snippets/spanner/spanner_create_instance_without_default_backup_schedule.go:27]: issue: provide a short Godoc-style comment that explains what this function does.

File: spanner/spanner_snippets/spanner/spanner_create_instance_without_default_backup_schedule.go [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+	// projectID := "my-project-id"
+	// instanceID := "my-instance"
+	ctx := context.Background()
+	instanceAdmin, err := instance.NewInstanceAdminClient(ctx)
+	if err != nil {
+		return err
+	}
+	defer instanceAdmin.Close()
+
+	op, err := instanceAdmin.CreateInstance(ctx, &instancepb.CreateInstanceRequest{
```
Comment (telpirion) [inline spanner/spanner_snippets/spanner/spanner_create_instance_without_default_backup_schedule.go:37]: issue(1): move the request object outside of the method call.

issue(2): provide a comment about the request object that describes the features of the resource you're creating.

**Review by telpirion — CHANGES_REQUESTED** [review]

**Review by telpirion — APPROVED** [review]

**Review by harshachinta — APPROVED** [review]

**Pull request merged**